### ISSUE TITLE
Delay code does not appear to be working

### ISSUE THREAD

**Issue by glennblock** [Jewelbots/arduino-library#11]
Given the following program

```
#include <Arduino.h>

LED led;
bool ran;

void setup() {
}

void loop() {
  if (!ran) {
    led.on(1, "blue", 100000);
    ran = true;    
  }
}
```

The blue light will blip and then instantly goes dark, there is no delay.

**Comment by glennblock** [comment]
I've noticed something which maybe matters (or maybe not).

In `nrf.c` the [type](https://github.com/Jewelbots/arduino-library/blob/master/jewelbots_nRF51822_board_v100/cores/JWB_nRF51822/ble-nrf51822-master/source/nordic_sdk/components/drivers_nrf/delay/nrf_delay.c#L19) declaration for `number_of_ms` is `uint32_t` while in LED.cpp, the [length](https://github.com/Jewelbots/arduino-library/blob/master/jewelbots_nRF51822_board_v100/cores/JWB_nRF51822/LED.cpp#L34) param is `uint8_t`

**Comment by glennblock** [comment]
Looks like the underlying `nrf_delay_us` function is not working. Try this code and both lights will be on continually.

```
#include <Arduino.h>

LED led;

void setup() {
  animation.rainbows();
}

void loop() {
  led.on(1, "magenta", 100000);
  uint32_t delay = 5000;
  nrf_delay_us(delay);
  led.on(2, "red", 100000);
  nrf_delay_us(delay);
}
```

**Comment by gortok** [comment]
5000 microseconds is not a very long time at all.

**Comment by glennblock** [comment]
Ah it is microseconds? I thought it was milli

**Comment by gortok** [comment]
nrf_delay_ms is milliseconds, it's in the same nrf_delay.h file.

**Comment by glennblock** [comment]
Ok, so in that case why have the LED light delay be microseconds? Wouldn't
milliseconds be more reasonable / intuitive?

I will try out using micro.

**Comment by glennblock** [comment]
These [docs](http://alpha.jewelbots.com/t/working-with-leds/66) are wrong then

```
Each LED on the Jewelbot is refferred to by a number 0 through 3, which are called pins (i.e. pin 0, pin 1, etc). There are 4 different LEDs on the Jewelbot to control.

There is a set amount of pre-made color addresses for the Jewelbot.
* orange
* blue
* pink
* red
* violet
* magenta
* green

Time is addressed in milliseconds (i.e. if you want to have the LED on for 2 seconds, you'd write 2000
```

**Comment by glennblock** [comment]
It doesn't appear it is working with microseconds either.

Given this code, I just see one color sometimes magenta, sometimes red.

```
#include <Arduino.h>

LED led;

void setup() {
}

void loop() {
  led.on(1, "magenta", 1000000);
  led.on(1, "red", 1000000);
}
```

**Comment by glennblock** [comment]
Looks like type conversion / variable layout or something is causing the problem.

If i run this code, I will see different color blips every second.

```
void loop() {
  uint32_t delay = 1000000;
  led.on(1, "magenta", 1);
  nrf_delay_us(delay);
  led.on(1, "red", 1);
  nrf_delay_us(delay);
}
```

This code however does not work properly while `led.on` is calling `nrf_delay_us()`, the same function. 

```
void loop() {
  uint32_t delay = 1000000;
  led.on(1, "magenta", delay);
  led.on(1, "red", delay);
}
```

**Comment by glennblock** [comment]
Fixed with #14
